### PR TITLE
feat(api): add JWT authentication

### PR DESCRIPTION
**Pull request by Alexandcoats**
Not sure how to handle password stuff. I figure the front end should hash it before sending over http, but this is a weak area for me.

### CODE REVIEW

File: bin/inx-chronicle/config.template.toml [hunk 1]
```diff
@@ -6,6 +6,7 @@ password = "root"
 [api]
 port = 8042
 allow_origins = ["0.0.0.0"]
+password_hash = "67ae144380e29fe9a4c8dff5d90de916"
```
Comment (Alex6323) [inline bin/inx-chronicle/config.template.toml:9]: Maybe add the default password as a comment.

File: bin/inx-chronicle/src/api/error.rs [hunk 2]
```diff
@@ -153,8 +153,9 @@ impl IntoResponse for ErrorBody {
 impl From<ApiError> for ErrorBody {
     fn from(err: ApiError) -> Self {
         if let ApiError::Internal(e) = &err {
-            log::error!("{}", e);
+            log::error!("Internal API error: {}", e);
```
Comment (pvdrz) [inline bin/inx-chronicle/src/api/error.rs:156]: Is it a good idea to log something inside `from`?

File: bin/inx-chronicle/src/api/routes.rs [hunk 3]
```diff
@@ -15,7 +22,47 @@ pub fn routes() -> Router {
...
+    )? {
+        let now = time::OffsetDateTime::now_utc();
+        let exp = now + config.jwt_expiration;
+        let jwt = jsonwebtoken::encode(
+            &Header::default(),
+            &serde_json::json!({
+                "iss": ApiData::ISSUER,
+                "sub": uuid::Uuid::new_v4().to_string(),
+                "aud": ApiData::AUDIENCE,
+                "nbf": (now - time::OffsetDateTime::UNIX_EPOCH).whole_seconds() as u64,
```
Comment (pvdrz) [inline bin/inx-chronicle/src/api/routes.rs:55]: I'd define these as variables for clarity and to deduplicate `nbf` and `iat`

File: bin/inx-chronicle/src/launcher.rs [hunk 4]
```diff
@@ -58,28 +64,57 @@ impl Actor for Launcher {
             .await;
 
         #[cfg(feature = "api")]
-        cx.spawn_child(super::api::ApiWorker::new(&db, &config.api)).await;
+        #[allow(clippy::redundant_clone)]
```
Comment (pvdrz) [inline bin/inx-chronicle/src/launcher.rs:67]: where's the redundant clone?

**Review by pvdrz — APPROVED** [review]

File: bin/inx-chronicle/src/api/error.rs (hunk 2 above)
Comment (Alexandcoats) [reply bin/inx-chronicle/src/api/error.rs:156]: Yes

File: bin/inx-chronicle/src/launcher.rs (hunk 4 above)
Comment (Alexandcoats) [reply bin/inx-chronicle/src/launcher.rs]: oop

**Review by Alex6323 — APPROVED** [review]

**Pull request merged**